rust/parquet/src/encodings/encoding.rs

```diff
@@ -130,6 +130,7 @@ impl<T: DataType> PlainEncoder<T> {
 }
 
 impl<T: DataType> Encoder<T> for PlainEncoder<T> {
+    #[cold]
```

Comment:
Performance Notes:

As far as I can tell these functions are rarely called and as such we can hint to the compiler that they dont need to be folded into hot locations in the final output.

Reply:
that might be helpful context to add as a comment in the source code so it outlasts this PR review